pkg/mmesh/modelmesh_events.go

```diff
@@ -176,7 +176,7 @@ func (mes *ModelMeshEventStream) refreshWatches(nw *namespaceWatch, namespace, s
 			}
 			if owner, err := ownerIDFromVModelRecord(value); err == nil {
 				encodedNamespace := namespace
-				if owner != "" {
+				if owner != "" && (len(namespace) <= len(owner) || owner+"_" != namespace[:len(owner)+1]) {
```

Comment:
@chinhuang007 this can also be reverted I think... should be no changes at all needed to this file.